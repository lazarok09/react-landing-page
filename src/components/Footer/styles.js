import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';
export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.font.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
      padding-top: 0;
      padding-bottom: 0;
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
