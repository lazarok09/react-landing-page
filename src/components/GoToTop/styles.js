import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 6;
    bottom: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    opacity: 0.8;
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
  `}
  &:hover {
    opacity: 1;
  }
`;
