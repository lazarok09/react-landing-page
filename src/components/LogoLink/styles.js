import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: inherit;
    display: flex;
    align-items: center;
    text-decoration: none;

    > img {
      max-height: 2.5rem;
    }
  `}
`;
