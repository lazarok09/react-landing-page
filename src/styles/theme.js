export const theme = {
  colors: {
    whiteBg: 'white',
    darkBg: 'black',
    colorRed: 'red',
  },
};
export default theme;
