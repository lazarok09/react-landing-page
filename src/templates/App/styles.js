import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.darkBg};
    color: ${theme.colors.colorRed};
  `}
`;
