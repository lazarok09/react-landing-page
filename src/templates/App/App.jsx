import * as Styled from './styles';

function App() {
  return (
    // se mudar a prop muda a cor lรก
    <Styled.Wrapper>
      <h1>Hello World</h1>;
    </Styled.Wrapper>
  );
}

export default App;
