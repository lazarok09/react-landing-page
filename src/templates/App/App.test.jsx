import { screen } from '@testing-library/react';
import App from './App';
import { renderTheme } from '../../styles/render-theme';
import theme from '../../styles/theme';

it('should render the App with theme provider', () => {
  const { debug } = renderTheme(<App />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Hello World',
  }).parentElement;
  expect(headingContainer).toHaveStyle(`
  background: ${theme.colors.darkBg}
  `);
  expect(headingContainer).toHaveStyleRule('background', 'black');
  expect(headingContainer).toMatchSnapshot();
});
