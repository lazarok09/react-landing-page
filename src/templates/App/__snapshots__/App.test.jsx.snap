// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App with theme provider 1`] = `
.c0 {
  background: black;
  color: red;
}

<div
  class="c0"
>
  <h1>
    Hello World
  </h1>
  ;
</div>
`;
